### ISSUE TITLE
Add PyPy2 and PyPy3 builds on Travis

### ISSUE THREAD

**Issue by ghackebeil** [Pyomo/pyomo#447]
...once expr_dev is merged. Also update README about support for PyPy once we confirm Travis tests pass.

**Comment by ghackebeil** [comment]
@whart222, @jsiirola: Here are some action items you can take that will allow me to start modifying the Travis builds to use our own custom docker images.

- Create an account on Docker Hub.
- Optionally create a Pyomo organization on Docker Hub
- Create an empty public Docker Hub repository with a name like pyomo-test-builds
- Create a public GitHub repo with a name like Pyomo/docker-builds (or some other name you prefer) and add a .travis.yml file, like what is below, that contains the travis-encrypted Docker Hub login credentials of whoever owns the Docker Hub repo.
```
sudo: required
cache: false
language: python
services:
- docker
env:
  global:
  - secure: ...
  - secure: ...
script:
  - echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
```
- Give me push access to master on the GitHub repo

Ping me once that is ready, and I'll starting working on it in the next few weeks.

**Comment by whart222** [comment]
I'll work on this soon.  Why do we need a separate github repo?  Won't we be able to do this from the Pyomo/pyomo repo?

**Comment by ghackebeil** [comment]
Yes, we _could_ do this from the Pyomo/pyomo repo. But note that:

 1. The build process can be somewhat lengthy (e.g., compiling CBC from source) and we don't want to further clog the build pipeline for Pyomo/pyomo.
 2. These builds shouldn't be triggered by pushes to Pyomo/pyomo, since there is no dependency. If anything, the dependency is reversed. That is, if we create new images (e.g., because there is a new version of some solver we are testing with), then that should likely trigger a new Pyomo/pyomo build on Travis.

What would go in the new GitHub repo would be the scripts that are used to build the docker images, which people could update when new solver versions come out. Pushing those changes to this new repo would trigger Travis to build and push them to Docker Hub (where the Travis would pull them from when testing the Pyomo/pyomo repo).

Make sense?

**Comment by ghackebeil** [comment]
this has been resolved as of #479
